storage: TestAcceptsUnsplitRanges nil pointer dereference

From [this build](https://circleci.com/gh/petermattis/cockroach/1500?utm_campaign=build-failed&utm_medium=email&utm_source=notification):

```
I160909 12:48:23.342062 606 gossip/gossip.go:230  initial resolvers: []
W160909 12:48:23.342101 606 gossip/gossip.go:1041  no resolvers found; use --join to specify a connected node
I160909 12:48:23.342196 606 gossip/gossip.go:264  NodeID set to 1
I160909 12:48:23.342293 606 storage/engine/rocksdb.go:356  opening in memory rocksdb instance
I160909 12:48:23.350019 606 gossip/gossip.go:264  NodeID set to 1
I160909 12:48:23.350176 606 gossip/gossip.go:264  NodeID set to 1
I160909 12:48:23.350852 636 storage/replica.go:2840  [s1,r1{/Min-/Max}] gossiping cluster id {"00000000-0000-0000-0000-000000000000"} from store 1, range 1
I160909 12:48:23.351290 628 storage/replica.go:2232  [s1,r1{/Min-/Max}] applying command with forced error: storage/replica.go:2157: no-op on empty Raft entry
E160909 12:48:23.351567 628 storage/replica.go:2267  [s1,r1{/Min-/Max}] error executing raft command: storage/replica.go:2157: no-op on empty Raft entry
I160909 12:48:23.352216 628 storage/replica_trigger.go:308  new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=1970-01-01 00:00:00 +0000 UTC]
I160909 12:48:23.352249 628 storage/replica.go:2862  gossiping sentinel from store 1, range 1
I160909 12:48:23.352295 628 storage/replica.go:2869  gossiping first range from store 1, range 1: [{1 1 1}]
I160909 12:48:23.352562 636 storage/replica.go:2862  [s1,r1{/Min-/Max}] gossiping sentinel from store 1, range 1
I160909 12:48:23.352622 636 storage/replica.go:2869  [s1,r1{/Min-/Max}] gossiping first range from store 1, range 1: [{1 1 1}]
I160909 12:48:23.355117 606 storage/queue.go:298  [test] [n1,s1,r3{/Table/11-/Max}]: split needed; not adding
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa73c0c]

goroutine 650 [running]:
panic(0x15bdf40, 0xc4200120d0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Recover(0xc420156090)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:176 +0x6e
panic(0x15bdf40, 0xc4200120d0)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/coreos/etcd/raft.(*raft).appendEntry(0xc42019e3c0, 0xc4202217c0, 0x1, 0x1)
	/go/src/github.com/coreos/etcd/raft/raft.go:465 +0xdc
github.com/coreos/etcd/raft.(*raft).becomeLeader(0xc42019e3c0)
	/go/src/github.com/coreos/etcd/raft/raft.go:551 +0x1d1
github.com/coreos/etcd/raft.(*raft).campaign(0xc42019e3c0, 0x170ede6, 0x10)
	/go/src/github.com/coreos/etcd/raft/raft.go:558 +0x709
github.com/coreos/etcd/raft.(*raft).Step(0xc42019e3c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/coreos/etcd/raft/raft.go:606 +0x243
github.com/coreos/etcd/raft.(*RawNode).Campaign(0xc42012bd80, 0x0, 0x0)
	/go/src/github.com/coreos/etcd/raft/rawnode.go:127 +0x71
github.com/cockroachdb/cockroach/storage.(*Replica).withRaftGroupLocked(0xc420151200, 0xc4205c1ae0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:416 +0x2a2
github.com/cockroachdb/cockroach/storage.defaultProposeRaftCommandLocked(0xc420151200, 0xc42007a3c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1637 +0x223
github.com/cockroachdb/cockroach/storage.(*Replica).proposePendingCmdLocked(0xc420151200, 0xc42007a3c0, 0xc420012530, 0xc4205cecc0)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1586 +0x77
github.com/cockroachdb/cockroach/storage.(*Replica).proposeRaftCommand(0xc420151200, 0x7f3da1c600f0, 0xc420012530, 0x0, 0x16, 0x0, 0x0, 0x3, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1565 +0x1f3
github.com/cockroachdb/cockroach/storage.(*pendingLeaseRequest).InitOrJoinRequest.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/replica_range_lease.go:133 +0x1a9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc420156090, 0x1dba3f1, 0x1e, 0xa8, 0xc4205be360)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:231 +0x95
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:232 +0xbc
```

@bdarnell Any idea what could have happened here? I'm going to see if this reliably fails.

--- a/storage/store_test.go
+++ b/storage/store_test.go
@@ -165,6 +165,13 @@ func createTestStoreWithoutStart(t testing.TB, ctx *StoreContext) (*Store, *hlc.
 
 func createTestStore(t testing.TB) (*Store, *hlc.ManualClock, *stop.Stopper) {
 	ctx := TestStoreContext()
+	// Many tests using this test harness (as opposed to higher-level
+	// ones like multiTestContext or TestServer) want to micro-manage
+	// replicas and the background queues just get in the way. The
+	// scanner doesn't run frequently enough to expose races reliably,
+	// so just disable the scanner for all tests that use this function
+	// instead of figuring out exactly which tests need it.
+	ctx.TestingKnobs.DisableScanner = true
 	return createTestStoreWithContext(t, &ctx)
 }
 
@@ -280,6 +287,11 @@ func TestBootstrapOfNonEmptyStore(t *testing.T) {
 	}
 }
 
+// create a Replica and add it to the store. Note that replicas
+// created in this way do not have their raft groups fully initialized
+// so most KV operations will not work on them. This function is
+// deprecated; new tests should create replicas by splitting from a
+// properly-bootstrapped initial range.
 func createReplica(s *Store, rangeID roachpb.RangeID, start, end roachpb.RKey) *Replica {
 	desc := &roachpb.RangeDescriptor{
 		RangeID:  rangeID,

